### PR TITLE
certcheck: tweaks and improvements

### PR DESCRIPTION
**Pull request by daviddrysdale**
Prob'ly best reviewed commit-by-commit

### CODE REVIEW

File: x509util/certcheck/certcheck.go [hunk 1]
```diff
@@ -59,38 +59,31 @@ func main() {
 
 	failed := false
 	for _, filename := range flag.Args() {
-		dataList, err := x509util.ReadPossiblePEMFile(filename, "CERTIFICATE")
+		chain, err := chainFromFile(filename)
 		if err != nil {
-			glog.Errorf("%s: Failed to read data: %v\n", filename, err)
+			glog.Errorf("%v\n", err)
```
Comment (AlCutter) [inline x509util/certcheck/certcheck.go]: [preexisting] I guess the `Errorf` calls through-out don't need the trailing `\n`

File: x509util/certcheck/certcheck.go [hunk 2]
```diff
@@ -95,6 +104,39 @@ func main() {
...
+	u, err := url.Parse(target)
+	if err != nil {
+		return nil, fmt.Errorf("%s: failed to parse URL: %v", target, err)
+	}
+	if u.Scheme != "https" {
+		return nil, fmt.Errorf("%s: non-https URL provided", target)
+	}
+	host := u.Host
+	if !strings.Contains(host, ":") {
+		host += ":443"
```
Comment (AlCutter) [inline x509util/certcheck/certcheck.go]: does `tls.Dial` not do the right thing by default, or is this about begin explicit for logging etc.?

**Comment by daviddrysdale** [comment]
Ping?

File: x509util/certcheck/certcheck.go (hunk 2 above)
Comment (AlCutter) [reply x509util/certcheck/certcheck.go]: d'oh, thanks

**Pull request merged**